### DISCUSSION TITLE
Feature request: add a flag to mark each bookmark public or private

### DISCUSSION THREAD

**Discussion by mfioretti** [go-shiori/shiori#331] [Feature requests]
IMHO, the two main reasons to self-host one's own bookmarks online are:

1. use them always in the same way, from any browser and device
2. **share** some of them with others, even if only in read-only mode

At the moment, shiori does not support #2, because **ALL** bookmarks are private, being only accessible after one logs in.

Can you please add a checkbox, and field in the database, to mark each bookmark  as public or private, so that the public ones:

1. are visible and searchable even to anonymous visitors
2. but they **never** link to the cached version, only to the original URL (needed both to save bandwidth on one's server **and** to avoid any copyright infringement issue (making a full **private** copy is one thing, putting online a full copy of some page that **everybody** can read is a totally different thing, legally speaking..)

Thanks

**Comment by Br31zh** [comment]
I’d love to see that feature too, the main reason I’m using a tool like this is tho share bookmarks, because for my own usage I’m using Firefox sync. I’m having a hard time finding out a good self-hosted link sharing software, and Shiori is the more promising.
